src/port1.0/portextract.tcl

```diff
@@ -165,7 +165,10 @@ proc portextract::extract_main {args} {
             set dir [file tail [lindex $workdirs 0]]
             ui_debug [format [msgcat::mc "Symlink: %s -> %s"] $worksrcpath $dir]
             symlink $dir $worksrcpath
+            set_source_date_epoch [lindex $workdirs 0]
         }
+    } else {
+        set_source_date_epoch $worksrcpath
```

Comment:
Does this work correctly if the `extract` and `destroot` steps are run separately, i.e. when I do

```
sudo port -v extract
sudo port -v destroot
```

It seems to me this would lead to `source_date_epoch` being left at 0. I guess we should record the value of `source_date_epoch` in the statefile, and read it from there in the other phases that need it.